content/client-lib-development-guide/features.textile

```diff
@@ -289,7 +289,7 @@ h3(#realtime-connection). Connection
 ** @(RTN15b)@ In order for a connection to be resumed and connection state to be recovered, the client library reconnects to the websocket endpoint with two additional querystring params:
 *** @(RTN15b1)@ @resume@ is the private connection key assigned to the connection when the first @CONNECTED@ @ProtocolMessage@ was received
 *** @(RTN15b2)@ @connection_serial@ is the most recent @ProtocolMessage#connectionSerial@ received from Ably or @Connection#serial@ which should be identical
-** @(RTN15c)@ When a new connection is made using the @resume@ and @connection_serial@ params, the client library must consider the resume to have failed if the @CONNECTED@ @ProtocolMessage@ contains an error or if the @Connection#id@ has changed. When a @Connection@ resume has failed, all channels should be detach with a suitable error reason
+** @(RTN15c)@ When a new connection is made using the @resume@ and @connection_serial@ params, the client library must consider the resume to have failed if the @CONNECTED@ @ProtocolMessage@ contains an error or if the @Connection#id@ has changed. When a @Connection@ resume has failed, all channels should be detached with a suitable error reason, triggering any messages awaiting an ACK to fail
```

Comment:
I think this is confused.

`ACK` and `NACK` responses for published messages can only ever be received on the transport connection on which those messages were sent. Therefore, once a transport drops, the client library must either fail the publish attempt, or re-attempt by re-sending the messages on a new transport if the resume was successful (ie the `CONNECTED` response includes the expected `connectionId`).

The system's response to a resume request can be a `CONNECTED` message with an `error`, if the error does not prevent the establishement of a connection. The outcome may be a resume of the requested connection (which implies that the connection state is intact, and all attached channels remain attached) but there may have been a non-fatal error (such as that the system is unable to send the backlog in full because the limit of backlog messages has been exceeded), or the outcome may be that a new connection is established (in which case the library must assume that all connection state is lost).